crates/cheatnet/tests/cheatcodes/prank.rs

```diff
@@ -9,22 +11,94 @@ use crate::{
+use cairo_felt::felt_str;
+use cairo_felt::Felt252;
 use cairo_lang_starknet::contract::starknet_keccak;
 use cheatnet::constants::build_testing_state;
+use cheatnet::constants::TEST_ADDRESS;
 use cheatnet::forking::state::ForkStateReader;
+use cheatnet::runtime_extensions::call_to_blockifier_runtime_extension::rpc::CallResult;
+use cheatnet::runtime_extensions::call_to_blockifier_runtime_extension::RuntimeState;
 use cheatnet::runtime_extensions::forge_runtime_extension::cheatcodes::declare::declare;
 use cheatnet::runtime_extensions::forge_runtime_extension::cheatcodes::spy_events::{
     Event, SpyTarget,
 };
-use cheatnet::state::{CheatTarget, CheatnetState, ExtendedStateReader};
+use cheatnet::state::{CheatSpan, CheatTarget, CheatnetState, ExtendedStateReader};
 use conversions::felt252::FromShortString;
 use conversions::IntoConv;
+use scarb_api::StarknetContractArtifacts;
 use starknet_api::block::BlockNumber;
 use starknet_api::core::ContractAddress;
+use starknet_api::core::{ClassHash, PatriciaKey};
+use starknet_api::hash::StarkHash;
+use starknet_api::patricia_key;
 use tempfile::TempDir;
 
+struct TestEnvironment<'a> {
+    cached_state: CachedState<ExtendedStateReader>,
+    runtime_state: RuntimeState<'a>,
+}
+
+impl<'a> TestEnvironment<'a> {
+    fn new(cheatnet_state: &'a mut CheatnetState) -> Self {
+        let cached_state = create_cached_state();
+        let runtime_state = build_runtime_state(cheatnet_state);
+        Self {
+            cached_state,
+            runtime_state,
+        }
+    }
+
+    fn declare(
+        &mut self,
+        contract_name: &str,
+        contracts: &HashMap<String, StarknetContractArtifacts>,
+    ) -> ClassHash {
+        let name = Felt252::from_short_string(contract_name).unwrap();
+        declare(&mut self.cached_state, &name, contracts).unwrap()
+    }
+
+    fn deploy(&mut self, contract_name: &str, calldata: &[Felt252]) -> ContractAddress {
+        deploy_contract(
+            &mut self.cached_state,
+            &mut self.runtime_state,
+            contract_name,
+            calldata,
+        )
+    }
+
+    fn call_contract(
+        &mut self,
+        contract_address: &ContractAddress,
+        selector: &str,
+        calldata: &[Felt252],
+    ) -> CallResult {
+        call_contract(
+            &mut self.cached_state,
+            &mut self.runtime_state,
+            contract_address,
+            &felt_selector_from_name(selector),
+            calldata,
+        )
+    }
+
+    fn start_prank(&mut self, target: CheatTarget, new_address: u128, span: CheatSpan) {
+        self.runtime_state.cheatnet_state.start_prank(
+            target,
+            ContractAddress::from(new_address),
+            span,
+        );
+    }
+
+    fn stop_prank(&mut self, contract_address: &ContractAddress) {
+        self.runtime_state
+            .cheatnet_state
+            .stop_prank(CheatTarget::One(*contract_address));
+    }
+}
+
```

Comment:
Something like this could be abstracted for other cheat tests, as they tend get pretty unreadable